modules/oe_webtools_media/tests/src/FunctionalJavascript/MediaSourceWebtoolsTest.php

```diff
@@ -102,6 +103,58 @@ public function testMediaWebtoolsSource(): void {
     }
   }
 
+  /**
+   * Tests the webtools media source support for wcloud.
+   */
+  public function testMediaWebtoolsSourceWcloud(): void {
+    $session = $this->getSession();
+    $page = $session->getPage();
+    $assert_session = $this->assertSession();
+
+    // Create a webtools media type for charts.
+    $this->createWebtoolsMediaType('test_media_webtools_type', 'chart');
+
+    // Create a webtools wcloud media item without a url.
+    $this->drupalGet("media/add/test_media_webtools_type");
+    $name = "Valid webtools wcloud item";
+    $assert_session->fieldExists('Name')->setValue($name);
+    $assert_session->fieldExists("Webtools Chart snippet")->setValue('{"utility": "wcloud"}');
+    $page->pressButton('Save');
+    $assert_session->pageTextContains("The provided wcloud url is not valid.");
+
+    // Create a webtools wcloud media item with an invalid url.
+    $assert_session->fieldExists("Webtools Chart snippet")->setValue('{"utility": "wcloud", "url": "not-a-url"}');
+    $page->pressButton('Save');
+    $assert_session->pageTextContains("The provided wcloud url is not valid.");
+
+    // Create a webtools wcloud media item with a url
+    // outside the europa.eu domain.
+    $assert_session->fieldExists("Webtools Chart snippet")->setValue('{"utility": "wcloud", "url": "https://google.com"}');
+    $page->pressButton('Save');
+    $assert_session->pageTextContains("The wcloud url needs to be in the europa.eu domain.");
+
+    // Create a webtools wcloud media item with a url that
+    // is not accessible/does not exist.
+    $assert_session->fieldExists("Webtools Chart snippet")->setValue('{"utility": "wcloud", "url": "https://europa.eu/fake-url"}');
+    $page->pressButton('Save');
+    $assert_session->pageTextContains("Could not parse contents of the wcloud url.");
+
+    // Create a webtools wcloud media item with a url that returns
+    // a correct response, but for the wrong widget type.
+    $assert_session->fieldExists("Webtools Chart snippet")->setValue('{"utility": "wcloud", "url": "https://europa.eu/correct-wcloud?widget=map"}');
+    $page->pressButton('Save');
+    $assert_session->pageTextContains("Invalid webtools chart snippet.");
+
+    // Create a webtools wcloud media item with a url that returns
+    // a correct response.
+    $assert_session->fieldExists("Webtools Chart snippet")->setValue('{"utility": "wcloud", "url": "https://europa.eu/correct-wcloud?widget=chart"}');
+    $page->pressButton('Save');
```

Comment:
After the save, assert the success message please to make it easy to debug that it fails then later if we have issues.